src/DataFormatter/JSONDataFormatter.php

```diff
@@ -118,10 +118,14 @@ public function convertDataObjectToJSONObject(DataObjectInterface $obj, $fields
                     ? $this->sanitiseClassName($relClass) . '/' . $obj->$fieldName
                     : $this->sanitiseClassName($className) . "/$id/$relName";
                 $href = Director::absoluteURL($rel);
-                $serobj->$relName = ArrayData::array_to_object(array(
+                $baseFields = [
                     "className" => $relClass,
                     "href" => "$href.json",
                     "id" => self::cast($obj->obj($fieldName))
+                ];
+                $serobj->$relName = ArrayData::array_to_object(array_replace(
+                    $baseFields,
+                    method_exists($serobj, 'getApiFields') ? (array) $serobj->getApiFields($baseFields) : []
```

Comment:
Do you want to use `hasMethod()` via ClassInfo here, which would also include extensions that provide this method?

Reply:
Nice...

Reply:
Done.